### PR TITLE
Add script to check for absl reference from Firestore/core/api

### PR DESCRIPTION
**Pull request by wu-hui**
Also add exception to check_imports.swift to exclude cmake build directories.

### CODE REVIEW

File: scripts/check_firestore_core_api_absl.py
Comment (var-const) [inline scripts/check_firestore_core_api_absl.py:75]: Nit: this comment repeats the message from the next line, I'd omit it.

File: scripts/check_firestore_core_api_absl.py
Comment (var-const) [inline scripts/check_firestore_core_api_absl.py:49]: Question: what is the use for dry run? From what I can see, the main difference is that it always returns a successful code upon exit.

File: scripts/check_firestore_core_api_absl.py
Comment (var-const) [inline scripts/check_firestore_core_api_absl.py:63]: Question: what does this command do?

**Review by var-const — CHANGES_REQUESTED** [review]
Overall looking good, just some nitpicks.

File: scripts/check_firestore_core_api_absl.py
Comment (wu-hui) [reply scripts/check_firestore_core_api_absl.py]: It just sets the log level to debug.

File: scripts/check_firestore_core_api_absl.py
Comment (wu-hui) [reply scripts/check_firestore_core_api_absl.py]: Done.

File: scripts/check_firestore_core_api_absl.py
Comment (wu-hui) [reply scripts/check_firestore_core_api_absl.py]: Done.

File: scripts/check_firestore_core_api_absl.py
Comment (wu-hui) [reply scripts/check_firestore_core_api_absl.py]: Done.

File: scripts/check_firestore_core_api_absl.py
Comment (wu-hui) [reply scripts/check_firestore_core_api_absl.py]: Done.

File: scripts/check_firestore_core_api_absl.py
Comment (wu-hui) [reply scripts/check_firestore_core_api_absl.py]: Done.

File: scripts/check_firestore_core_api_absl.py
Comment (wu-hui) [reply scripts/check_firestore_core_api_absl.py]: Done.

File: scripts/check_firestore_core_api_absl.py
Comment (wu-hui) [reply scripts/check_firestore_core_api_absl.py:75]: Done.

File: scripts/check_firestore_core_api_absl.py
Comment (var-const) [inline scripts/check_firestore_core_api_absl.py:46]: Nits:
- `s/usaging/usage/`;
- please use the `patterns` variable instead of hardcoding `api/` here.

**Review by var-const — COMMENTED** [review]
Question: is this possible to make it so that if this script fails, the failure is visible but the PR can still be merged? It's ok to add Abseil to the private section of a class in `api/`, but this script would still fail because it doesn't distinguish between `public` and `private` (and trying to do that would be more trouble than it's worth, IMO).

**Comment by var-const** [comment]
Mostly LGTM, but need to resolve the question about whether this check blocks/should block merging.

File: scripts/check_firestore_core_api_absl.py
Comment (wu-hui) [reply scripts/check_firestore_core_api_absl.py:46]: Done.

**Comment by wu-hui** [comment]
> Question: is this possible to make it so that if this script fails, the failure is visible but the PR can still be merged? It's ok to add Abseil to the private section of a class in `api/`, but this script would still fail because it doesn't distinguish between `public` and `private` (and trying to do that would be more trouble than it's worth, IMO).

We could change the call with `--dry-run` in `check.sh`. But we can wait until such a case did happen: when someone tried to add absl in private. I doubt this would happen before we remove this check.

**Review by var-const — APPROVED** [review]

**Comment by var-const** [comment]
@wu-hui Can you please do a test PR that adds Abseil to the `api` folder after you merge this? Just to make sure that the check fails and to take a look at the script output.

**Comment by var-const** [comment]
@wu-hui Andy, is this ready to merge? Please let me know if you need anything from me.

**Pull request merged**
